core/src/main/java/org/ldaptive/ssl/X509CertificatesCredentialReader.java

```diff
@@ -1,24 +1,46 @@
 /* See LICENSE for licensing and NOTICE for copyright. */
 package org.ldaptive.ssl;
 
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.security.GeneralSecurityException;
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 /**
- * Loads X.509 certificate credentials from a classpath, filepath, or stream resource. Supported certificate formats
- * include: PEM, DER, and PKCS7.
+ * Loads X.509 certificate credentials from a classpath, filepath, or stream resource.
+ * When working with filepath, multiple files may be separated using a comma (i.e. {@code cert1.pem,cert2.crt}).
+ * Supported certificate formats include: PEM, DER, and PKCS7.
  *
  * @author  Middleware Services
  */
 public class X509CertificatesCredentialReader extends AbstractCredentialReader<X509Certificate[]>
 {
 
-
+  @Override
+  public X509Certificate[] read(final String path, final String... params)
+      throws IOException, GeneralSecurityException
+  {
+    final String[] paths = path.split(",");
+    return Arrays.stream(paths)
+        .map(individualPath -> {
+          try {
+            return Arrays.asList(super.read(individualPath, params));
+          } catch (final Exception e) {
+            logger.warn(String.format("Unable to read certificate at %s", individualPath), e);
```

Comment:
I want this exception to bubble out. Especially in the case of reading trust material.

Reply:
Because this is handled inside a lambda, I could either wrap the exception inside a RuntimeException and throw back or some other unchecked exception, or shy away from using the lambda and instead just use a normal for-loop to collect certificates. What would you recommend?

Reply:
I would avoid the lambda so you can deliver an `IOException`.